lib/plugins/login/login.js

```diff
@@ -0,0 +1,160 @@
+      readlineInterface.question('Please enter the verification code here: ', (code) => {
+        const authorizationData = {
+          code,
+          code_verifier: verifier,
+          client_id: config.AUTH0_CLIENT_ID,
+          grant_type: 'authorization_code',
+          redirect_uri: config.AUTH0_CALLBACK_URL,
+        };
+        // verify login
+        fetch(`${config.AUTH0_URL}/oauth/token`, {
+          method: 'POST',
+          body: JSON.stringify(authorizationData),
+          headers: { 'content-type': 'application/json' },
+        })
+          .then((response) => response.json())
+          .then((platformResponse) => {
+            const decoded = jwtDecode(platformResponse.id_token);
+            // console.log('decoded', decoded);
+            this.serverless.cli.log('You are now logged in');
+
+            // because platform only support github
+            const id = decoded.original_user_id || decoded.sub;
+
+            /* For future use
+            segment.identify({
+              userId: id,
+              traits: {
+                email: profile.email,
+              },
+            }) */
+
+            const userConfig = {
+              userId: id,
+              frameworkId,
+              users: {},
+            };
+            // set user auth in global .serverlessrc file
+            userConfig.users[id] = {
+              userId: id,
+              name: decoded.name,
+              email: decoded.email,
+              auth: platformResponse,
+            };
+
+            // update .serverlessrc
+            setConfig(userConfig);
+
+            const userID = new Buffer(id).toString('base64');
+            const email = new Buffer(decoded.email).toString('base64');
+            const name = new Buffer(decoded.name).toString('base64');
+            const loginCount = decoded.login_count;
+            const createdAt = decoded.created_at;
+            const successUrl = `https://serverless.com/success?u=${userID}&e=${email}&n=${name}&c=${loginCount}&v=${version}&d=${createdAt}&id=${frameworkId}`; // eslint-disable-line
+
+            openBrowser(successUrl);
+            // identify user for better onboarding
+            userStats.identify({
+              id,
+              frameworkId,
+              email: decoded.email,
+              // unix timestamp
+              created_at: Math.round(+new Date(createdAt) / 1000),
+            }).then(() => {
+              userStats.track('user_loggedIn', {
+                id,
+                email: decoded.email,
+              }).then(() => {
+                // console.log('login complete');
+                // then exit process
+                process.exit(0);
+              });
+            });
+          })
+          .catch(() => {
+            console.log(chalk.red('Incorrect token value supplied. Please run `serverless login` again')); // eslint-disable-line
```

Comment:
Maybe we can wrap `serverless login` into "" since the console output won't render markdown (the tildes).